src/adapters/poi_popup.js

```diff
@@ -0,0 +1,67 @@
+import {Popup} from 'mapbox-gl--ENV'
+import OsmSchedule from '../../src/adapters/osm_schedule'
+import Poi from '../mapbox/poi'
+import IconManager from "./icon_manager";
+const poiSubClass = require('../mapbox/poi_subclass')
+let popupTemplate = require('../views/popup.dot')
+const poiConfigs = require('../../config/constants.yml').pois
+
+const WAIT_BEFORE_DISPLAY = 800
+
+
+function PoiPopup() {}
+
+PoiPopup.prototype.init = function(map) {
+  this.map = map
+  this.popupHandle = null
+
+  this.timeOutHandler = null
+  this.isCounting = false
+}
+
+PoiPopup.prototype.addListener = function(layer) {
+  this.map.on('mouseenter', layer, (e) => {
+    this.isCounting = true
+    this.timeOutHandler = setTimeout(() => {
+      let poi = e.features[0]
+      this.create(poi)
+    }, WAIT_BEFORE_DISPLAY)
+  })
+
+  this.map.on('mouseleave', layer, async () => {
+    this.isCounting = false
+    clearTimeout(this.timeOutHandler)
+  })
+}
+
+PoiPopup.prototype.create = async function (layerPoi) {
+  let poi = await Poi.poiApiLoad(layerPoi.properties.global_id)
+  if(poi) {
+    let {color} = IconManager.get(poi)
```

Comment:
This breaks when the `poi` has no specific icon, (and `IconManager.get` returns undefined).  
We need a default color in that case.